alacritty/src/event.rs

```diff
@@ -554,14 +554,23 @@ impl<N: Notify + OnResize> Processor<N> {
                     },
                     WindowEvent::KeyboardInput { input, is_synthetic: false, .. } => {
                         processor.key_input(input);
-                        if input.state == ElementState::Pressed {
-                            // Hide cursor while typing
-                            if processor.ctx.config.ui_config.mouse.hide_when_typing {
-                                processor.ctx.window.set_mouse_visible(false);
-                            }
+                        // Hide cursor while typing, when triggering a keybind that writes to pty
+                        if input.state == ElementState::Pressed
+                            && processor.ctx.config.ui_config.mouse.hide_when_typing
+                            && *processor.ctx.received_count() > 0
```

Comment:
This is not what `received_count` is for and it will break the existing `received_char` method.

Why is this not done directly in `input.rs`?